R/pkg/R/types.R

```diff
@@ -88,11 +88,6 @@ specialtypeshandle <- function(type) {
 checkSchemaInArrow <- function(schema) {
   stopifnot(inherits(schema, "structType"))
 
-  requireNamespace1 <- requireNamespace
```

Comment:
I checked all the usages of `checkSchemaInArrow`:

```
grep -Fnr "checkSchemaInArrow" R
R/types.R:88:checkSchemaInArrow <- function(schema) {
R/SQLContext.R:277:        checkSchemaInArrow(schema)
R/group.R:235:      checkSchemaInArrow(schema)
R/DataFrame.R:1211:                checkSchemaInArrow(schema(x))
R/DataFrame.R:1509:      checkSchemaInArrow(schema)
```

These are all within branches that have checked

```
arrowEnabled <- sparkR.conf("spark.sql.execution.arrow.sparkr.enabled")[[1]] == "true"
```

Since `arrow` is not directly used and this conf check is passed already, I think the `requireNamespace` here is unnecessary.

Reply:
In the meantime, given that `sparkR.conf("spark.sql.execution.arrow.sparkr.enabled")[[1]] == "true"` was popping up in a few places, I added a helper to `utils.R` to centralize maintenance of that snippet.

Reply:
Oh, this was a fail-fast to show a better error message :-).

Reply:
OK, reverted.